### PR TITLE
NEX-150: Updates to Drupal code, contrib and next.js to 14.2.2

### PR DESCRIPTION
**Pull request by vermario**
## Link to ticket:

https://wunder.atlassian.net/browse/NEX-150

## Link to feature environment:

https://nex-150-next.next-drupal-starterkit.dev.wdr.io/

## Changes proposed in this PR:

This PR updates: 

* Drupal core to 10.2.5
* Contrib modules to their latest versions. The most notable being:
   * graphql module to 4.7
   * graphql_compose to beta6, which includes a way to return the langcode for menu items so we can remove our local crusty patch :-)
* Next.js to the 14.2.2 version: https://github.com/vercel/next.js/releases/tag/v14.2.2   

This also includes an improvement for the pRetry functionality, where we set the retries to 0 while in development mode to avoid making the feedback loop too long while developing for example graphql queries locally.

### CODE REVIEW

File: drupal/composer.json [hunk 1]
```diff
@@ -139,9 +140,6 @@
             "drupal/graphql" : {
                 "drupal-graphql#1323: Add check for translation (from Github)": "https://patch-diff.githubusercontent.com/raw/drupal-graphql/graphql/pull/1388.patch"
             },
-            "drupal/graphql_compose" : {
-                "Add the langcode field to the menu_item entity in graphql (local patch)": "./patches/grapqhl_compose_add_langcode_to_menu_item.patch"
-            },
```
Comment (vermario) [inline drupal/composer.json:144]: No longer needed after this was added to graphql_compose https://git.drupalcode.org/project/graphql_compose/-/commit/12d3d19ef0c78827da673622441917a384f9ba14

File: drupal/recipes/wunder_next_setup/config/graphql.graphql_servers.graphql_compose_server.yml
Comment (vermario) [inline drupal/recipes/wunder_next_setup/config/graphql.graphql_servers.graphql_compose_server.yml:1]: Some changes due to graphql module new version

File: drupal/recipes/wunder_next_setup/config/graphql_compose.settings.yml
Comment (vermario) [inline drupal/recipes/wunder_next_setup/config/graphql_compose.settings.yml:1]: Some changes due to graphql_compose module new version

File: next/components/footer/footer.tsx [hunk 2]
```diff
@@ -15,7 +15,9 @@ interface FooterProps {
 export function Footer({ menu }: FooterProps) {
   // Only show the menu items that match the current locale:
   const { locale } = useRouter();
-  const filteredItems = menu.items?.filter((link) => link.langcode == locale);
+  const filteredItems = menu.items?.filter(
+    (link) => link.langcode?.id == locale,
+  );
```
Comment (vermario) [inline next/components/footer/footer.tsx:20]: The new version of graphql_compose introduces the `langcode` as a properly shaped `language` object, so we have to use langcode.id here

File: next/lib/drupal/drupal-client.ts [hunk 3]
```diff
@@ -14,7 +14,7 @@ const getFetcher =
     };
     // Return the pRetry wrapped fetch function:
     return pRetry(wrappedFetch, {
-      retries: 5,
+      retries: env.NODE_ENV === "development" ? 0 : 5,
```
Comment (vermario) [inline next/lib/drupal/drupal-client.ts:17]: 0 retries in dev mode, means the error will be returned immediately

File: next/lib/drupal/graphql-drupal-client.ts [hunk 4]
```diff
@@ -23,7 +25,7 @@ export class GraphQlDrupalClient extends DrupalClient {
     return await pRetry(
       () => request(endpoint, query, variables, requestHeaders),
       {
-        retries: 5,
+        retries: env.NODE_ENV === "development" ? 0 : 5,
```
Comment (vermario) [inline next/lib/drupal/graphql-drupal-client.ts:28]: 0 retries in dev mode, means the error will be returned immediately

File: next/lib/graphql/queries.ts [hunk 5]
```diff
@@ -77,35 +77,41 @@ export const GET_MENU = graphql(`
...
-        langcode
         title
         internal
         attributes {
           class
           icon
         }
+        langcode {
+          id
+        }
```
Comment (vermario) [inline next/lib/graphql/queries.ts:88]: new shape of the langcode property in menus

File: .lando.yml [hunk 6]
```diff
@@ -187,6 +187,17 @@ services:
...
+  #
+  # kibana:
+  #   type: lando
+  #   services:
+  #     image: "docker.elastic.co/kibana/kibana:7.17.0"
+  #     command: "/bin/tini -- /usr/local/bin/kibana-docker"
+  #     user: kibana
+  #     ports:
+  #       - "5601:5601"
+
```
Comment (vermario) [inline .lando.yml:200]: Some people might still want to use kibana, so we can add it back as a commented out service (FYI @misterjoonas )

Comment (badrange) [reply .lando.yml:200]: An idea: You could put this in a `lando.local.yml.example` file; if someone wants kibana they can rename it to `.lando.local.yml` without ending up with a .lando.yml file that isn't in sync with the repo

Comment (vermario) [reply .lando.yml:200]: hi @badrange ! 

the thing is that this is not really meant to be updated after you have used it once to kickstart your project

Comment (badrange) [reply .lando.yml:200]: Ah, right, makes sense!

**Review by Zionknight-crypto — APPROVED** [review]
Set up process was smooth and no issues to report as well as no console errors. Frontend and Backend appear to be working as expected.

File: .lando.yml [hunk 7]
```diff
@@ -187,6 +187,17 @@ services:
       environment:
         ELASTICVUE_CLUSTERS: '[{"name": "dev cluster", "uri": "http://localhost:9200", "password": "", "user": ""}]'
 
+  # Uncomment this section if you prefer to use the mode advanced Kibana tool with Elasticsearch:
```
Comment (joshua-scott) [inline .lando.yml:190]:
```suggestion
  # Uncomment this section if you prefer to use the more advanced Kibana tool with Elasticsearch:
```

**Review by joshua-scott — APPROVED** [review]
Tested by running ./setup.sh locally, everything worked smoothly 👍

**Pull request merged**